### PR TITLE
Adding validation rule to check for x-ms-examples

### PR DESCRIPTION
**Pull request by dsgouda**
Addresses issue https://github.com/Azure/autorest/issues/1954

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/Validation/XmsExamplesProvidedValidation.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        /// </summary>
+        /// <remarks>
+        /// This may contain placeholders '{0}' for parameterized messages.
+        /// </remarks>
+        public override string MessageTemplate => "Please provide x-ms-examples describing minimum/maximum property set for response/request payloads for operation '{0}' ";
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Error;
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/XmsExamplesProvidedValidation.cs]: @salameer @kirthik could you confirm if you'd like this rule as an error or a warning?

File: src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs [hunk 2]
```diff
@@ -555,6 +555,18 @@ public void BodyParametersReferencedValidation()
             var messages = ValidateSwagger(Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "Validation", "body-parameters-validation-2.json"));
             messages.AssertOnlyValidationMessage(typeof(BodyParametersValidation), 1);
         }
+
+
+        /// <summary>
+        /// Verifies that sku object
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs]: update comment

File: src/modeler/AutoRest.Swagger/Validation/XmsExamplesProvidedValidation.cs (hunk 1 above)
Comment (kirthik) [inline src/modeler/AutoRest.Swagger/Validation/XmsExamplesProvidedValidation.cs]: @salameer  and I discussed this and this should be an error.

**Comment by olydis** [comment]
@dsgouda ...but fix the failing tests ;)

**Comment by veronicagg** [comment]
After talking to Samer, as this will cause lots of messages, we'd like to do the following: if there are more than 10 issues, show only 1 message that indicates that "x-ms-examples" are missing for more than 10 operations, and point to "paths" as the path of the issue. If there are less than 10 errors, then show the appropriate messages, pointing to each of the corresponding operations/places where the rule is violated. 
If this gets too complicated to implement, let's chat. thanks!

File: src/modeler/AutoRest.Swagger/Validation/XmsExamplesProvidedValidation.cs (hunk 1 above)
Comment (salameer) [inline src/modeler/AutoRest.Swagger/Validation/XmsExamplesProvidedValidation.cs]: Yep it should be an error.

One possiblity that veronica and I just discussed is:

- If we have less that ten (or another threshold) we should show all the errors to help the user track.
- if we have more than the threshold it should be (one Generic error + 10 errors to help the user navigate)

if this is too complicated we can show just one  generic (but the user wouldn't have swiggles)

let me know

**Pull request merged**